pkg/cli/root_test.go

```diff
@@ -1018,11 +1018,56 @@ func TestVerify(t *testing.T) {
 	})
 }
 
+func TestAutoPortSelection(t *testing.T) {
+	Convey("Run server without specifying a port (port 0)", t, func() {
+		logFile, err := os.CreateTemp("", "zot-log*.txt")
+		logPath := logFile.Name()
+		So(err, ShouldBeNil)
+
+		content := fmt.Sprintf(`{"distSpecVersion": "1.0.0", "storage": {"rootDirectory": "/tmp/zot"},
+		"http": {"address": "127.0.0.1", "port": "0"},
+		"log": {"level": "debug", "output": "%s"}}`, logFile.Name())
+
+		cfgfile, err := os.CreateTemp("", "zot-test*.json")
+		So(err, ShouldBeNil)
+
+		defer os.Remove(cfgfile.Name()) // clean up
+
+		_, err = cfgfile.Write([]byte(content))
+		So(err, ShouldBeNil)
+
+		os.Args = []string{"cli_test", "serve", cfgfile.Name()}
+		go func() {
+			err = cli.NewServerRootCmd().Execute()
+			So(err, ShouldBeNil)
+		}()
+
+		_, err = ReadLogFileAndSearchString(logPath, "HTTP port", 90*time.Second)
```

Comment:
this function return err only if it can't read the log file, the other return param is whether or not the string was found.
please check that the other param is true.